### PR TITLE
Configure linter to catch British 🇬🇧  spellings 🇺🇸 🦅 📖 

### PR DESCRIPTION
**Pull request by strideynet**
I've noticed that I tend to use British 🇬🇧  spellings of various words across the Teleport codebase (clearly a mistake made in a momentary lapse of focus as I dip my biscuit in my tea 🍵 ).

I was curious what putting `misspell` into the US 🦅  localisation 🌎  would do, and if there would be any breaking changes made to the code 😡 . I've raised this PR to make it easier for me to explore the diff, and I may eventually raise this as a full PR for merging 🥺 once I'm satisfied this doesn't introduce any risk.

There's two words I'm a little unsure of 😕 :
- Cancelled/canceled ? I'm pretty sure `canceled` is the Americanized version, but there seems to be heavy usage of `cancelled` in the codebase already. The Go stdlib (e.g `context.Canceled`) seems to use the `canceled` spelling which is an argument in favour.
- Marshalled/marshaled ? Again, I'm fairly certain `marshaled` would be the correct American spelling, and this matches several things in the Go stdlib (e.g `json.Marshal`), but we seem to use `marshalled` in many places.

### CODE REVIEW

File: lib/services/database.go [hunk 1]
```diff
@@ -702,7 +702,7 @@ func IsRedshiftClusterAvailable(cluster *redshift.Cluster) bool {
 	// assumed to be unavailable if the cluster cannot be created or restored.
 	switch aws.StringValue(cluster.ClusterStatus) {
 	case "available", "available, prep-for-resize", "available, resize-cleanup",
-		"cancelling-resize", "final-snapshot", "modifying", "rebooting",
+		"canceling-resize", "final-snapshot", "modifying", "rebooting",
```
Comment (strideynet) [inline lib/services/database.go:705]: This value must remain as it is.

**Comment by strideynet** [comment]
I think I'm pretty happy this doesn't break anything, but I'm not 100% sure this won't be too annoying as a developer experience. I'm curious what others think so I'll raise this as a PR to get some feedback on it.

**Review by zmb3 — APPROVED** [review]

**Review by Joerger — APPROVED** [review]
This reminds me of a candidate coding challenge where I "corrected" a bunch of typos without realizing they were just the british spellings :sweat_smile: ... anyways LGTM

**Review by espadolini — APPROVED** [review]
`americanise-spellings` huh

**Comment by strideynet** [comment]
> `americanise-spellings` huh

This may have been intentional. God save the queen 👑

**Comment by greedy52** [comment]
revisiting this after three years. 

I have to use `nolint` for `cancelled` unfortunately (https://github.com/gravitational/teleport/pull/59773/commits/7966e8b9b9f70827cda29666068eea6473e6a286) as it comes from the mcp spec 😂 :
- https://modelcontextprotocol.io/specification/2025-03-26/basic/utilities/cancellation

**Pull request merged**